### PR TITLE
Wren: add package

### PR DESCRIPTION
**Pull request by burdockcascade**
This adds the Wren scripting language to the repo.

**Links**
- https://github.com/wren-lang/wren
- https://wren.io/

**Popularity**
- Over 6k stars on Github
- over 500 forks on Github

### CODE REVIEW

File: packages/w/wren/xmake.lua [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+                if is_mode("debug") then
+                    add_defines("DEBUG")
+                elseif is_mode("release") then
+                    add_defines("NDEBUG")
+                end
+
+        ]], package:config("shared") and "shared" or "static"))
+        if package:config("shared") then
+            configs.kind = "shared"
+        end
```
Comment (qudix) [inline packages/w/wren/xmake.lua:31]: This isn't necessary, xmake already passes kind

File: packages/w/wren/xmake.lua [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+                set_kind(kind)
+                add_headerfiles("src/include/*.h", "src/vm/*.h", "src/optional/*.h")
+                add_includedirs("src/include", "src/vm", "src/optional")
+                add_files("src/vm/*.c", "src/optional/*.c")
+
+                if is_mode("debug") then
+                    add_defines("DEBUG")
+                elseif is_mode("release") then
+                    add_defines("NDEBUG")
+                end
```
Comment (qudix) [inline packages/w/wren/xmake.lua:26]: The `mode.release` rule already sets `NDEBUG`, but `mode.debug` does not set `DEBUG` which I'm unsure is common

File: packages/w/wren/xmake.lua [hunk 3]
```diff
@@ -0,0 +1,79 @@
+package("wren")
+    set_homepage("http://wren.io")
+    set_description("The Wren Programming Language. Wren is a small, fast, class-based concurrent scripting language.")
```
Comment (qudix) [inline packages/w/wren/xmake.lua:3]: Shorten description by removing the first sentence so it's just:
`Wren is a small, fast, class-based concurrent scripting language.`

File: packages/w/wren/xmake.lua (hunk 1 above)
Comment (burdockcascade) [reply packages/w/wren/xmake.lua]: fixed

File: packages/w/wren/xmake.lua (hunk 3 above)
Comment (burdockcascade) [reply packages/w/wren/xmake.lua]: fixed

File: packages/w/wren/xmake.lua (hunk 2 above)
Comment (burdockcascade) [reply packages/w/wren/xmake.lua]: fixed

File: packages/w/wren/xmake.lua [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+                    add_defines("DEBUG")
+                end
+
+        ]], package:config("shared") and "shared" or "static"))
+        import("package.tools.xmake").install(package, configs)
+    end)
+
+    on_test(function (package)
+        assert(package:check_csnippets({test = [[
+            int main() {
```
Comment (c8ef) [inline packages/w/wren/xmake.lua:32]: Consider using `void test` instead of `int main`.

**Review by c8ef — CHANGES_REQUESTED** [review]

File: packages/w/wren/xmake.lua [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+                    
+                    result = wrenCall(vm, 0); 
+
+                    if (result != WREN_RESULT_SUCCESS) {
+                        fprintf(stderr, "Error calling main function.\n");
+                        wrenFreeVM(vm);
+                        return 1; 
+                    }
+                } else {
+                    fprintf(stderr, "Error interpreting Wren script.\n");
```
Comment (waruqi) [inline packages/w/wren/xmake.lua:63]: too large test snippet, we can simply ot. we need only to test api call and links. we can use has_cfuns to check it

Comment (burdockcascade) [reply packages/w/wren/xmake.lua]: fixed

File: packages/w/wren/xmake.lua (hunk 4 above)
Comment (burdockcascade) [reply packages/w/wren/xmake.lua]: this test has now been simplified

**Pull request merged**